recipes/tcl/8.6.10/conanfile.py

```diff
@@ -0,0 +1,222 @@
+            with tools.chdir(self.build_folder):
+                autoTools = self._get_auto_tools()
+                autoTools.make(target="install-binaries")
+                autoTools.make(target="install-libraries")
+                autoTools.make(target="install-msgs")
+                autoTools.make(target="install-tzdata")
+                autoTools.make(target="install-headers")
+                autoTools.make(target="install-private-headers")
+            pkgconfig_dir = os.path.join(self.package_folder, "lib", "pkgconfig")
+            tools.rmdir(pkgconfig_dir)
+        self.copy(pattern="license.terms", dst="licenses", src=self._source_subfolder)
+
+        tclConfigShPath = os.path.join(self.package_folder, "lib", "tclConfig.sh")
+        package_path = os.path.join(self.package_folder)
+        if self._is_mingw_windows:
+            package_path = package_path.replace("\\", "/")
+        tools.replace_in_file(tclConfigShPath,
+                              package_path,
+                              "${TCL_ROOT}")
+        tools.replace_in_file(tclConfigShPath,
+                              "\nTCL_BUILD_",
+                              "\n#TCL_BUILD_")
+
+        tools.replace_in_file(tclConfigShPath,
+                              "\nTCL_SRC_DIR",
+                              "\n#TCL_SRC_DIR")
+
+    def package_info(self):
+        libs = []
+        systemlibs = []
+        libdirs = []
+        for root, _, _ in os.walk(os.path.join(self.package_folder, "lib"), topdown=False):
+            newlibs = tools.collect_libs(self, root)
+            if newlibs:
+                libs.extend(newlibs)
+                libdirs.append(root)
+        if self.settings.os == "Windows":
+            systemlibs.extend(["ws2_32", "netapi32", "userenv"])
+        else:
+            systemlibs.extend(["m", "pthread", "dl"])
+
+        defines = []
+        if not self.options.shared:
+            defines.append("STATIC_BUILD")
+        self.cpp_info.defines = defines
+
+        self.cpp_info.bindirs = ["bin"]
+        self.cpp_info.libdirs = libdirs
+        self.cpp_info.libs = libs
+        self.cpp_info.system_libs = systemlibs
+        self.cpp_info.includedirs = ["include"]
+
+        if self.settings.os == "Macos":
+            self.cpp_info.frameworks = ["Cocoa"]
+            self.cpp_info.sharedlinkflags = self.cpp_info.exelinkflags
+
+        tcl_library = os.path.join(self.package_folder, "lib", "{}{}".format(self.name, ".".join(self.version.split(".")[:2])))
+        self.output.info("Setting TCL_LIBRARY environment variable to {}".format(tcl_library))
+        self.env_info.TCL_LIBRARY = tcl_library
+
+        tcl_root = self.package_folder
+        self.output.info("Setting TCL_ROOT environment variable to {}".format(tcl_root))
+        self.env_info.TCL_ROOT = tcl_root
+
+        tclsh_list = list(filter(lambda fn: fn.startswith("tclsh"), os.listdir(os.path.join(self.package_folder, "bin"))))
+        assert(len(tclsh_list))
```

Comment:
Why do you need `assert`? Is this not deterministic?

Reply:
I originally added this as a fail-safe while developing.
I guess it is not needed.